app/src/main/java/ai/elimu/appstore/BaseApplication.java

```diff
@@ -99,4 +121,53 @@ public Retrofit getRetrofit() {
 
         return retrofit;
     }
+
+    /**
+     * Provides {@SecurePreferences} singleton instance to access shared preferences
+     *
+     * @return A single instance of {@SecurePreferences}
+     */
+    public static SharedPreferences getSharedPreferences() {
+        if (sSecurePreferences == null) {
+            synchronized (BaseApplication.class) {
+                if (sSecurePreferences == null) {
+                    sSecurePreferences = new SecurePreferences(getAppContext(), PREF_PASSWORD,
```

Comment:
I suggest replacing `PREF_PASSWORD` with `getKeyHash()`, and removing the `PREF_PASSWORD` variable. That way, the key hash won't be called until it's actually needed, and the amount if code is reduced.

Reply:
Firstly, getKeyHash() needs to access a non-static method: getPackageName().
Therefore it cannot be static.

Secondly, the method getSharedPreferences() needs to be static as I explained in previous comment

Because getSharedPreferences() is using returned value from getKeyHash(), we need to create an intermediate static variable, which is PREF_PASSWORD.